_posts/2016-11-29-future.md

```diff
@@ -0,0 +1,206 @@
+
+- All plugins should have examples/repl links. Can also embed runkit.
+- Update Learn ES2015 to have ES2015+.
+- Add resources page
+- Update FAQ with common errors
+- Page for community plugins
+- babel-handbook, API docs, etc
+
+## Future
+
+### [Plugin API Changes](https://github.com/babel/notes/blob/master/2016-08/august-01.md#potential-api-changes-for-traversal)
+
+Currently many users have trouble with plugin/preset ordering and it's very confusing what plugin is run before/after and how to make it do so. This is a fundamental change to the core of Babel which will take some work to figure out.
+
+### Use Type Information
+
+Integrate with a type system such as flow/typescript to make optimizations.
+
+> Sidenote: Be able to understand typescript syntax (similar to flow), add a plugin to strip typescript types for better interop.
+
+### Read the dependency graph
+
+Integrate with other tools like webpack/bundlers? This will allow cross file transformations or whole codebase optimizations.
+
+### Parser Errors
+
+Better error messages from the parser like in [Compiler Errors for Humans](http://elm-lang.org/blog/compiler-errors-for-humans).
+
+<blockquote class="twitter-tweet" data-lang="en"><p lang="en" dir="ltr">babel-eslint@7.1.1: now adds the code frame when there&#39;s a parser error! <a href="https://t.co/yoxRpGXq5E">pic.twitter.com/yoxRpGXq5E</a></p>&mdash; Henry Zhu (@left_pad) <a href="https://twitter.com/left_pad/status/799388723896946692">November 17, 2016</a></blockquote>
+<script async src="//platform.twitter.com/widgets.js" charset="utf-8"></script>
+
+> It's obvious that we all want to see helpful errors!
+
+- [babel/babylon#125](https://github.com/babel/babylon/pull/125): Instead of "Unexpected token" explain that `await` needs to be used in an async function.
+- Do better inference/guessing on user intention to prevent vague errors.
+
+### `babel-init`
+
+Setup Babel more easily for those still fatigued.
+
+- Setup a .babelrc from scratch
+- Reuse the `babel` npm package to default to use `babel-preset-env`
+- Can possibly do a similar thing to `create-react-app` with ejecting to `babel-cli` + `.babelrc`
+
+```sh
+npm install babel
+# prompt and setup preset-env
+./node_modules/.bin/babel-init
+```
+
+### Run [tc39/test262](https://github.com/tc39/test262)
+
+Running official spec tests against Babel to make sure we comply to the spec or at least know when we don't.
+
+### Smoke Tests
+
+[motiz88/babel-smoke-tests](https://github.com/motiz88/babel-smoke-tests).
+
+### Program Analysis
+
+- [Program Slicing](https://en.wikipedia.org/wiki/Program_slicing) with [slice-js](http://slides.com/kentcdodds/faster-javascript#/)
+- Semantic (AST aware) Grepping tool
+
+### `babel --debug`
+
+Print out all info when erroring. Also perf tooling to get more info on how long each plugin takes.
+
+### Syntax plugins:
+
+Opening the gates might lead to an ecosystem of incompatible plugins.
```

Comment:
Not sure what you're hinting at with opening the gates and an ecosystem of incompatible plugins. Are there any examples?